corehq/apps/locations/models.py

```diff
@@ -107,13 +111,20 @@ def _populate_stock_levels(self):
         self.overstock_threshold = config.overstock_threshold
 
     def save(self, *args, **kwargs):
+        from .tasks import sync_administrative_status
         if not self.code:
             from corehq.apps.commtrack.util import unicode_slug
             self.code = unicode_slug(self.name)
         if not self.commtrack_enabled:
             self.administrative = True
         self._populate_stock_levels()
-        return super(LocationType, self).save(*args, **kwargs)
+        saved = super(LocationType, self).save(*args, **kwargs)
+
+        if self._administrative_old != self.administrative:
```

Comment:
should we add `and self._administrative_old is not None` here to avoid spinning up celery tasks when locations are first created?